### DISCUSSION TITLE
Create the NPM package

### DISCUSSION THREAD

**Discussion by bobboteck** [bobboteck/JoyStick#16] [Ideas]
@bartbutenaers do you have any guide that I can follow, to create the NPM package?

**Comment by bartbutenaers** [comment]
@bobboteck,

For my Github repositories, I did it simply like this:
1. I create a package.json file (like e.g. [this ](https://github.com/bartbutenaers/node-red-contrib-msg-speed/blob/master/package.json)one), which contains:
   + mostly a series of links to my Github repository.  
   + If your code has dependencies to other npm packages, you can add those in the `dependencies` section.  That way they will be downloaded automatically when I install your package.
   + The most tricky part is the entry point.  For me that is: 
   ```
    "node-red"     : {
        "nodes": {
            "msg-speed": "msg_speed.js"
        }
    }
    ```
   When I `require` my package, the entrypoint gives information about where the code will start.  So for you that part will be different.  I'm now doubting whether you should add ```"main": "joy.js",``` instead.  

2. When that file is ok, I publish a new version on npm (after I have done once `npm login` on this machine) using `npm publish https://github.com/bobboteck/JoyStick/tarball/master` (and of course update the version number of your package first in your package.json file).

I'm going to ask somebody else from the Node-RED community, whether he can join this discussion...

**Reply by bartbutenaers** [reply]
I have discussed this, and the `"main": "joy.js"`` should indeed be the way to go ...

But if the npm package gives you too much headache, I can also copy your library into my package.  It is not that big of a deal!!

**Reply by bobboteck** [reply]
Hi @bartbutenaers, is not a big problem, I need only the time to create the json file, I think to do it today

**Reply by bobboteck** [reply]
And now we have package:
https://www.npmjs.com/package/html5-joystick

**Reply by bartbutenaers** [reply]
Cool...  Thanks a lot for your time!!!
And of course I hope you have a happy new year evening!
Bart
